core/3d/CCBundle3DData.h

```diff
@@ -39,8 +39,135 @@
+        : _stride(formatToStride(backend::IndexFormat::U_INT)), _buffer(rhs)
+    {}
+
+    IndexArray(const IndexArray& rhs) : _stride(rhs._stride), _buffer(rhs._buffer) {}
+    IndexArray(IndexArray&& rhs) noexcept : _stride(rhs._stride), _buffer(std::move(rhs._buffer)) {}
+
+    IndexArray& operator=(const IndexArray& rhs)
+    {
+        _stride = rhs._stride;
+        _buffer = rhs._buffer;
+        return *this;
+    }
+    IndexArray& operator=(IndexArray&& rhs) noexcept
+    {
+        this->swap(rhs);
+        return *this;
+    }
+
+    void swap(IndexArray& rhs)
+    {
+        std::swap(_stride, rhs._stride);
+        _buffer.swap(rhs._buffer);
+    }
+
+    /** Clears the internal byte buffer. */
+    void clear() { _buffer.clear(); }
+
+    /** Pushes back a value. */
+    void push_back(uint32_t val)
+    {
+        assert(_stride == 2 || _stride == 4);
+        _buffer.append_n((uint8_t*)&val, _stride);
+    }
+
+    /** Inserts a list containing unsigned short (uint16_t) data. */
+    void insert(size_t offset, std::initializer_list<uint16_t> ilist, uint16_index_format /*U_SHORT*/)
+    {
+        assert(_stride == 2);
+        binsert(offset * _stride, ilist.begin(), ilist.end());
+    }
+
+    /** Inserts a list containing unsigned int (uint32_t) data. */
+    void insert(size_t offset, std::initializer_list<uint32_t> ilist, uint32_index_format /*U_INT*/)
+    {
+        assert(_stride == 4);
+        binsert(offset * _stride, ilist.begin(), ilist.end());
+    }
+
+    /** Inserts range data based on an offset in bytes. */
+    void binsert(size_t offset, const void* first, const void* last)
+    {
+        _buffer.insert(offset, (const uint8_t*)first, (const uint8_t*)last);
+    }
+
+    template <typename _Ty>
+    _Ty& at(size_t idx)
+    {
+        assert(sizeof(_Ty) == _stride);
+        if (idx < this->size())
+            return (_Ty&)_buffer[idx * sizeof(_Ty)];
+        throw std::out_of_range("IndexArray: out of range!");
+    }
+
+    uint8_t* data() noexcept { return _buffer.data(); }
+    const uint8_t* data() const noexcept { return _buffer.data(); }
+
+    /** returns the count of indices in the container. */
+    size_t size() const { return _buffer.size() / _stride; }
+    /** returns the size of the container in bytes. */
+    size_t bsize() const { return _buffer.size(); }
+
+    /** resizes the count of indices in the container. */
+    void resize(size_t size) { _buffer.resize(size * _stride); }
+    /** resizes the container in bytes. */
+    void bresize(size_t size) { _buffer.resize(size); }
+
+    /** returns true if the container is empty. Otherwise, false. */
+    bool empty() const { return _buffer.empty(); }
+
+    /** returns the format of the index array. */
+    backend::IndexFormat format() const { return strideToFormat(_stride); }
+
+    /** clears the buffer and sets the format specified. */
+    void format(backend::IndexFormat format = backend::IndexFormat::U_SHORT)
```

Comment:
Use `clear` maybe better because the major behavior is clear the internal byte buffer